### ISSUE TITLE
Cannot read nested Result types on client side

### ISSUE THREAD

**Issue by Dzoukr** [Zaid-Ajaj/Fable.Remoting#101]
In shared project, I have DU for errors:

```fsharp
type DomainError =
    | ItemAlreadyExists of Text.TextValue

type ServerError =
    | DomainError of DomainError

type ServerResponse<'a> = Async<Result<'a, ServerError>>
```

and proxy defined as:

```fsharp
type API = {
    Register : Account -> ServerResponse<Guid>
}
```

On error (in Result), the payload is serialized to:

```{"Error":{"DomainError":{"ItemAlreadyExists":"Email"}}}```

But when handling on client side, the automatic deserialization fails with:

```
Cannot convert ["JObject",{"comparer":{},"tree":["MapOne","Error",["JObject",{"comparer":{},"tree":["MapOne","DomainError",["JObject",{"comparer":{},"tree":["MapOne","ItemAlreadyExists",["JString","Email"]]}]]}]]}] to ["Any",null]
```

Is it because of nested union types or there is problem with transfering Result types as such?

Thanks a lot for help

**Comment by Zaid-Ajaj** [comment]
Hello there @Dzoukr, do you happen to use the latest Fable compiler? @alfonsogarciacaro said he broke something with reflection in the `Result<'a>` type, see [this issue](https://github.com/fable-compiler/Fable/issues/1655) 

It should have been fixed in v2.1.2, can you please check? 

In any case, remoting can handle nested unions without problems, see [protocol definition](https://github.com/Zaid-Ajaj/tabula-rasa/blob/master/Shared/Types.fs#L123) of tabula-rasa which looks very similar to yours

**Comment by alfonsogarciacaro** [comment]
Yes, please try with 2.1.2 or higher, sorry for the trouble. Please ping me if it's still not working.
